File: 0070-climbing-stairs/0070-climbing-stairs.py
class Solution:
    dic = {}
    def climbStairs(self, n: int) -> int:
        if n == 1 or n == 2:
            return n
        if n in self.dic:
            return self.dic[n]
        
        self.dic[n] = self.climbStairs(n-1) + self.climbStairs(n-2)
        return self.dic[n]

File: 0070-climbing-stairs/0070-climbing-stairs.java
class Solution {
    public HashMap<Integer, Integer> memo = new HashMap<Integer, Integer>();

    public int climbStairs(int n) {
        if(n == 1 || n ==2){
            return n;
        }
        if(memo.get(n) != null){
            return memo.get(n);
        }
        memo.put(n, climbStairs(n - 1) + climbStairs(n-2));
        return memo.get(n);
    }
}